src/Roslyn.Diagnostics.Analyzers/UnitTests/CSharpAvoidOptSuffixForNullableEnableCodeTests.cs

```diff
@@ -264,5 +265,99 @@ public void Method1({|CS0246:Class2|}? s)
 }",
             }.RunAsync();
         }
+
+        [Fact, WorkItem(3813, "https://github.com/dotnet/roslyn-analyzers/issues/3813")]
+        public async Task RS0046_CSharp8_NullableEnabledCode_InterfaceImplementation_NoDiagnostic()
+        {
+            await new VerifyCS.Test
+            {
+                LanguageVersion = LanguageVersion.CSharp8,
+                TestCode = @"
+#nullable enable
+
+public interface ISomething
+{
+    void Method1(string? [|sOpt|]);
+}
+
+public class Class1 : ISomething
+{
+    public void Method1(string? sOpt)
+    {
+        string? [|localOpt|] = null, [|otherLocalOpt|] = null;
+
+        System.Console.WriteLine(""{0}, {1}, {2}"", sOpt, localOpt, otherLocalOpt);
+    }
+}",
+                FixedCode = @"
+#nullable enable
+
+public interface ISomething
+{
+    void Method1(string? s);
+}
+
+public class Class1 : ISomething
+{
+    public void Method1(string? sOpt)
```

Comment:
Would it be better to still report if the name is different from the parent? In this context that would mean that after the fix, we would start reporting on `sOpt`as the interface definition is now `s`. On this note, do you have any analyzer that reports when a parameter name is changed between definitions (interface says s but class says sOpt)? If not, would you be interested in such analyzer?

Reply:
@mavasani Any opinion on this?

Reply:
>  On this note, do you have any analyzer that reports when a parameter name is changed between definitions (interface says s but class says sOpt)? If not, would you be interested in such analyzer?

I am not sure, @sharwell are you aware if we have such a naming rule in IDE code style naming analyzers? I am assuming this will likely go into Roslyn repo as a naming analyzer.